packages/legend-application-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx

```diff
@@ -450,6 +431,131 @@ const ProjectDependencyEditor = observer(
   },
 );
 
+const PlatformBasicEditor = observer(
+  (props: {
+    platform: ProjectServerPlatform;
+    projectConfig: ProjectConfiguration;
+    isReadOnly: boolean;
+  }) => {
+    const { platform, projectConfig, isReadOnly } = props;
+
+    const changeVersion = action((val: string | undefined): void => {
+      projectConfig.changePlatformVersion(platform, val ? val : '');
+    });
+
+    const isNotValidPlatformVersion = action(
+      (val: string): boolean => val === '',
+    );
+
+    return (
+      <PanelFormTextEditorWithValidation
+        isReadOnly={isReadOnly}
+        value={platform.platformVersion}
+        name={platform.name}
+        update={changeVersion}
+        validationErrorMessage={
+          isNotValidPlatformVersion(platform.platformVersion)
+            ? 'Version cannot be empty'
+            : undefined
+        }
+      />
+    );
+  },
+);
+
+const ProjectPlatformVersionEditor = observer(
+  (props: { projectConfig: ProjectConfiguration; isReadOnly: boolean }) => {
+    const { projectConfig, isReadOnly } = props;
+    const editorStore = useEditorStore();
+
+    const platformConfigurations = projectConfig.platformConfigurations;
+
+    if (platformConfigurations) {
+      editorStore.projectConfigurationEditorState.setOverrideDefaultPlatform(
+        true,
+      );
+    }
+
+    const isEditingPlatformConfigs =
+      editorStore.projectConfigurationEditorState.overridesDefaultPlatform;
+
+    const defaultPlatforms = editorStore.sdlcServerClient.platforms;
+
+    const overridePlatforms = (): void => {
+      if (isEditingPlatformConfigs !== true) {
+        editorStore.setActionAlertInfo({
+          message:
+            'This is an advanced configuration option meant for users who already have in mind the platform version they would like to input.',
```

Comment:
I think we need to go through what we should say here in details. Also I think it's good idea to give people some idea about this feature even before they click open this